charts/drax/charts/dashboard/templates/configmap-server.yaml

```diff
@@ -11,38 +11,46 @@ top:
 
 name: {{ include "accelleran.common.fullname" (dict "top" $) }}-server
 
+{{- $config := .Values.config }}
+
+# Define required variables
+{{- $requiredVariables := list
+  "nodeApiURL"
+  "nodeApiPort"
+  "kafkaURL"
+  "kafkaPort"
+  "networkStateMonitorHost"
+  "networkStateMonitorPort"
+  "svcMonitorHost"
+  "svcMonitorPort"
+  "svcOrchestratorHost"
+  "svcOrchestratorPort"
+  "configApiHost"
+  "configApiPort"
+  "pcixAppPodName"
+  "pcixAppPort"
+  "ksqldbPodName"
+  "ksqldbPort"
+  "LteRadioControllerLabel"
+}}
+
+
+{{- range $index, $key := $requiredVariables }}
+  {{- if not (index $config $key) }}
+    {{- fail (printf "Requires %s to be defined in values.yaml" $key) }}
+  {{- end }}
+{{- end }}
+
+
+# Template serviceName if not provided in configuration
+{{- $serviceName := include "accelleran.common.service.name" (dict "top" $) }}
+{{- $config := merge $config (dict "serviceName" $serviceName) }}
```

Comment:
I'm thinking if we can write this down in a clearer way?
Now someone reading this needs to understand the `merge` function (and not misunderstand it as `mergeOverwrite` for example).

Reply:
An option is to explicitly use `mergeOverwrite` with the operators reversed. Perhaps that is more clear?

Reply:
I'm more thinking about an if statement in combination with `set`: https://helm.sh/docs/chart_template_guide/function_list/#set

Reply:
```suggestion
{{- if (get $config "serviceName") }}
{{- $serviceName := include "accelleran.common.service.name" (dict "top" $) }}
{{- $_ := set $config "serviceName" $serviceName }}
{{- end }}
```

Reply:
Shouldn't that `if` be negated? If `serviceName` is empty, it will be falsey, and in that scenario, we do want to overwrite it.

Reply:
Yes indeed, was being to quick...